orchestration/dagster/hca_orchestration/resources/base.py

```diff
@@ -0,0 +1,165 @@
+})
+def local_beam_runner(init_context):
+    return LocalBeamRunner(working_dir=init_context.resource_config["working_dir"])
+
+
+class LocalBeamRunner:
+    def __init__(self, working_dir):
+        self.working_dir = working_dir
+
+    def run(self, job_name, input_prefix, output_prefix, context):
+        context.log.info("Local beam runner")
+        # TODO this is hardcoded to the HCA transformation pipeline for now
+        subprocess.run(
+            ["sbt", f'hca-transformation-pipeline/run --inputPrefix={input_prefix} --outputPrefix={output_prefix}'],
+            check=True,
+            cwd=f"{self.working_dir}"
+        )
+
+
+@resource({
+    "project": Field(StringSource),
+    "temp_location": Field(StringSource),
+    "subnet_name": Field(StringSource),
+    "service_account": Field(StringSource),
+    "image_name": Field(StringSource),
+    "image_version": Field(StringSource),
+    "namespace": Field(StringSource)
+})
+def dataflow_beam_runner(init_context):
+    return DataflowBeamRunner(
+        project=init_context.resource_config['project'],
+        temp_location=init_context.resource_config['temp_location'],
+        subnet_name=init_context.resource_config['subnet_name'],
+        service_account=init_context.resource_config['service_account'],
+        image_name=init_context.resource_config['image_name'],
+        image_version=init_context.resource_config['image_version'],
+        namespace=init_context.resource_config['namespace']
+    )
+
+
+class DataflowBeamRunner:
+    def __init__(self, project, temp_location, subnet_name, service_account, image_name, image_version, namespace):
+        self.project = project
+        self.temp_location = temp_location
+        self.subnet_name = subnet_name
+        self.service_account = service_account
+        self.image_name = image_name
+        self.image_version = image_version
+        self.namespace = namespace
+
+    def run(self, job_name, input_prefix, output_prefix, context):
+        args = [
+            '--runner=dataflow',
+            f"--inputPrefix={input_prefix}",
+            f"--outputPrefix={output_prefix}",
+            f"--project={self.project}",
+            f"--region=us-central1",
+            f"--tempLocation={self.temp_location}",
+            f"--subnetwork=regions/us-central1/subnetworks/{self.subnet_name}",
+            f"--serviceAccount={self.service_account}",
+            f"--workerMachineType=n1-standard-4",
+            f"--autoscalingAlgorithm=THROUGHPUT_BASED",
+            f"--numWorkers=4",
+            f"--maxNumWorkers=16",
+            f"--experiments=shuffle_mode=service"
+        ]
+
+        image_name = f"{self.image_name}:{self.image_version}"  # {context.solid_config['version']}"
+        job = self.dispatch_k8s_job(self.namespace, image_name, job_name, args, context)
+        context.log.info(f"Dataflow job started")
+
+        self.get_job_status(job.metadata.name, self.namespace)
```

Comment:
nit: prefer to call static methods as class methods, i.e. `DataflowBeamRunner.get_job_status(...)`, instead of instance methods